{
 "metadata": {
  "name": "",
  "signature": "sha256:800ff152db7c0263e4579071706e04ac1971151603a0c6a48a23f40d0ea8fd70"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "## Analysis of differences among S. mediterranea strains from different labs"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### Data used\n",
      "\n",
      "Available S. mediterranea raw data was downloaded from NCBI and organized (`ngsData-download.ipynb`). An index file (`data.info`) was made to describe each `.fastq` file. "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import sys, os\n",
      "from collections import Counter\n",
      "#import custom functions for displaying tables, bash commands\n",
      "sys.path.append(os.path.abspath(\"/home/damian/\"))\n",
      "from dk_ipython import *\n",
      "\n",
      "dataInfo = [line.strip().split('\\t') for line in open('/home/share/ngsData/smed/data.info')]\n",
      "fastqs = Counter([x.strip().split('_')[0] for x in os.listdir('/home/share/ngsData/smed/fastq')])\n",
      "illum = dict([(x[0],x) for x in dataInfo if x[9].lower().find('illumina') != -1])\n",
      "\n",
      "print 'Number of illumina sequenced libraries:', len(illum)\n",
      "print 'Number of labs:', len(set([x[2] for x in dataInfo]))\n",
      "print 'Labs:'\n",
      "print '\\n'.join(set(['  ' + x[2] for x in dataInfo]))\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Number of illumina sequenced libraries: 59\n",
        "Number of labs: 7\n",
        "Labs:\n",
        "  Bartscherer\n",
        "  Rajewsky\n",
        "  Graveley\n",
        "  Pearson\n",
        "  Sanchez\n",
        "  Rink\n",
        "  Newmark\n"
       ]
      }
     ],
     "prompt_number": 1
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### Trimming reads\n",
      "\n",
      "All Illumina reads were trimmed with Trimmomatic. Minimum length set at 30."
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#generate trimmmomatic commands and run with:\n",
      "#parallel :::: trimm.cmd > trimm.out 2> trimm.err \n",
      "trimCmdSEA = 'java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 '\n",
      "trimCmdPEA = 'java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 '\n",
      "trimCmdB = ' LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33'\n",
      "\n",
      "trimmomaticBash = ''\n",
      "for lib, count in fastqs.items():\n",
      "    if lib in illum:\n",
      "        fulllib = '/home/share/ngsData/smed/fastq/' + lib\n",
      "        phred = detectPhred(fulllib + '_1.fastq.gz', 5000) #in dk_ipython\n",
      "        if phred == 'Sanger':\n",
      "            phred = '-phred33 '\n",
      "        else:\n",
      "            phred = '-phred64 '\n",
      "        if count == 1:\n",
      "            #single end\n",
      "            trimmomaticBash += trimCmdSEA + phred + fulllib + '_1.fastq.gz ' + lib + '_1.trimmed.fastq.gz ' + trimCmdB + '\\n'\n",
      "        else:\n",
      "            #paired end\n",
      "            trimmomaticBash += trimCmdPEA + phred + fulllib + '_1.fastq.gz ' + fulllib + '_2.fastq.gz ' + \\\n",
      "            lib + '_1.paired.trimmed.fastq.gz ' + lib + '_1.unpaired.trimmed.fastq.gz ' + \\\n",
      "            lib + '_2.paired.trimmed.fastq.gz ' + lib + '_2.unpaired.trimmed.fastq.gz ' + trimCmdB + '\\n'\n",
      "            \n",
      "HTML(textarea('Trimmomatic bash commands', trimmomaticBash))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<b>Trimmomatic bash commands</b>:<br>\n",
        "<textarea style=\"width: 1000px; height:200px\">\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR900849_1.fastq.gz SRR900849_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR900848_1.fastq.gz SRR900848_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred64 /home/share/ngsData/smed/fastq/oenals5_1.fastq.gz /home/share/ngsData/smed/fastq/oenals5_2.fastq.gz oenals5_1.paired.trimmed.fastq.gz oenals5_1.unpaired.trimmed.fastq.gz oenals5_2.paired.trimmed.fastq.gz oenals5_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred64 /home/share/ngsData/smed/fastq/oenals4_1.fastq.gz /home/share/ngsData/smed/fastq/oenals4_2.fastq.gz oenals4_1.paired.trimmed.fastq.gz oenals4_1.unpaired.trimmed.fastq.gz oenals4_2.paired.trimmed.fastq.gz oenals4_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR900847_1.fastq.gz SRR900847_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR900846_1.fastq.gz SRR900846_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR388708_1.fastq.gz SRR388708_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR388711_1.fastq.gz SRR388711_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR388701_1.fastq.gz SRR388701_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR388703_1.fastq.gz SRR388703_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR388705_1.fastq.gz /home/share/ngsData/smed/fastq/SRR388705_2.fastq.gz SRR388705_1.paired.trimmed.fastq.gz SRR388705_1.unpaired.trimmed.fastq.gz SRR388705_2.paired.trimmed.fastq.gz SRR388705_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/ERR032066_1.fastq.gz /home/share/ngsData/smed/fastq/ERR032066_2.fastq.gz ERR032066_1.paired.trimmed.fastq.gz ERR032066_1.unpaired.trimmed.fastq.gz ERR032066_2.paired.trimmed.fastq.gz ERR032066_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR388712_1.fastq.gz SRR388712_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/ERR032069_1.fastq.gz /home/share/ngsData/smed/fastq/ERR032069_2.fastq.gz ERR032069_1.paired.trimmed.fastq.gz ERR032069_1.unpaired.trimmed.fastq.gz ERR032069_2.paired.trimmed.fastq.gz ERR032069_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/ERR032068_1.fastq.gz /home/share/ngsData/smed/fastq/ERR032068_2.fastq.gz ERR032068_1.paired.trimmed.fastq.gz ERR032068_1.unpaired.trimmed.fastq.gz ERR032068_2.paired.trimmed.fastq.gz ERR032068_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR390350_1.fastq.gz SRR390350_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR867382_1.fastq.gz SRR867382_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR867383_1.fastq.gz SRR867383_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR867380_1.fastq.gz SRR867380_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR867381_1.fastq.gz SRR867381_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR867386_1.fastq.gz SRR867386_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR867387_1.fastq.gz SRR867387_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR867384_1.fastq.gz SRR867384_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR867385_1.fastq.gz SRR867385_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR390352_1.fastq.gz SRR390352_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR390353_1.fastq.gz SRR390353_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR390351_1.fastq.gz SRR390351_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR125365_1.fastq.gz /home/share/ngsData/smed/fastq/SRR125365_2.fastq.gz SRR125365_1.paired.trimmed.fastq.gz SRR125365_1.unpaired.trimmed.fastq.gz SRR125365_2.paired.trimmed.fastq.gz SRR125365_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR125366_1.fastq.gz SRR125366_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR496276_1.fastq.gz /home/share/ngsData/smed/fastq/SRR496276_2.fastq.gz SRR496276_1.paired.trimmed.fastq.gz SRR496276_1.unpaired.trimmed.fastq.gz SRR496276_2.paired.trimmed.fastq.gz SRR496276_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR654079_1.fastq.gz SRR654079_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR1012832_1.fastq.gz SRR1012832_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR955511_1.fastq.gz /home/share/ngsData/smed/fastq/SRR955511_2.fastq.gz SRR955511_1.paired.trimmed.fastq.gz SRR955511_1.unpaired.trimmed.fastq.gz SRR955511_2.paired.trimmed.fastq.gz SRR955511_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR955099_1.fastq.gz /home/share/ngsData/smed/fastq/SRR955099_2.fastq.gz SRR955099_1.paired.trimmed.fastq.gz SRR955099_1.unpaired.trimmed.fastq.gz SRR955099_2.paired.trimmed.fastq.gz SRR955099_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred64 /home/share/ngsData/smed/fastq/oenalx2_1.fastq.gz /home/share/ngsData/smed/fastq/oenalx2_2.fastq.gz oenalx2_1.paired.trimmed.fastq.gz oenalx2_1.unpaired.trimmed.fastq.gz oenalx2_2.paired.trimmed.fastq.gz oenalx2_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR900852_1.fastq.gz SRR900852_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR900853_1.fastq.gz SRR900853_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR496278_1.fastq.gz /home/share/ngsData/smed/fastq/SRR496278_2.fastq.gz SRR496278_1.paired.trimmed.fastq.gz SRR496278_1.unpaired.trimmed.fastq.gz SRR496278_2.paired.trimmed.fastq.gz SRR496278_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR900851_1.fastq.gz SRR900851_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR1012834_1.fastq.gz SRR1012834_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR1012835_1.fastq.gz SRR1012835_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred64 /home/share/ngsData/smed/fastq/oenalx1_1.fastq.gz /home/share/ngsData/smed/fastq/oenalx1_2.fastq.gz oenalx1_1.paired.trimmed.fastq.gz oenalx1_1.unpaired.trimmed.fastq.gz oenalx1_2.paired.trimmed.fastq.gz oenalx1_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR900850_1.fastq.gz SRR900850_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR1012833_1.fastq.gz SRR1012833_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/ERR032067_1.fastq.gz /home/share/ngsData/smed/fastq/ERR032067_2.fastq.gz ERR032067_1.paired.trimmed.fastq.gz ERR032067_1.unpaired.trimmed.fastq.gz ERR032067_2.paired.trimmed.fastq.gz ERR032067_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR388706_1.fastq.gz SRR388706_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred64 /home/share/ngsData/smed/fastq/SRR654077_1.fastq.gz SRR654077_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR1012838_1.fastq.gz SRR1012838_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR867388_1.fastq.gz SRR867388_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/ERR032070_1.fastq.gz /home/share/ngsData/smed/fastq/ERR032070_2.fastq.gz ERR032070_1.paired.trimmed.fastq.gz ERR032070_1.unpaired.trimmed.fastq.gz ERR032070_2.paired.trimmed.fastq.gz ERR032070_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/ERR032071_1.fastq.gz /home/share/ngsData/smed/fastq/ERR032071_2.fastq.gz ERR032071_1.paired.trimmed.fastq.gz ERR032071_1.unpaired.trimmed.fastq.gz ERR032071_2.paired.trimmed.fastq.gz ERR032071_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/ERR032072_1.fastq.gz /home/share/ngsData/smed/fastq/ERR032072_2.fastq.gz ERR032072_1.paired.trimmed.fastq.gz ERR032072_1.unpaired.trimmed.fastq.gz ERR032072_2.paired.trimmed.fastq.gz ERR032072_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR388710_1.fastq.gz /home/share/ngsData/smed/fastq/SRR388710_2.fastq.gz SRR388710_1.paired.trimmed.fastq.gz SRR388710_1.unpaired.trimmed.fastq.gz SRR388710_2.paired.trimmed.fastq.gz SRR388710_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR1012785_1.fastq.gz SRR1012785_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR1012836_1.fastq.gz SRR1012836_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred64 /home/share/ngsData/smed/fastq/oenalxins_1.fastq.gz /home/share/ngsData/smed/fastq/oenalxins_2.fastq.gz oenalxins_1.paired.trimmed.fastq.gz oenalxins_1.unpaired.trimmed.fastq.gz oenalxins_2.paired.trimmed.fastq.gz oenalxins_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar PE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR496280_1.fastq.gz /home/share/ngsData/smed/fastq/SRR496280_2.fastq.gz SRR496280_1.paired.trimmed.fastq.gz SRR496280_1.unpaired.trimmed.fastq.gz SRR496280_2.paired.trimmed.fastq.gz SRR496280_2.unpaired.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR388713_1.fastq.gz SRR388713_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "java -jar /home/share/software/Trimmomatic-0.32/trimmomatic-0.32.jar SE -threads 1 -phred33 /home/share/ngsData/smed/fastq/SRR1012831_1.fastq.gz SRR1012831_1.trimmed.fastq.gz  LEADING:20 TRAILING:20 MINLEN:30 ILLUMINACLIP:/home/share/software/Trimmomatic-0.32/adapters/adapters.fa:2:30:12:1:false TOPHRED33\n",
        "\n",
        "</textarea>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 2,
       "text": [
        "<IPython.core.display.HTML at 0x1b86110>"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### Trimmomatic output"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#load trimming log file (trimm.err) and check\n",
      "trimmLib = {}\n",
      "prev = ''\n",
      "trimmLog = open('/home/share/projects/smed_strain/trimm/trimm.err')\n",
      "for line in trimmLog:\n",
      "    if line.find('Started with arguments') != -1:\n",
      "        prev = line.strip().split('/fastq/')[1].split('_1.fastq.gz')[0]\n",
      "    if line.find('Surviving') != -1:\n",
      "        meta = line.strip().split(':')\n",
      "        numReads = int(meta[1].split()[0])\n",
      "        keys = ['Reads'] + [meta[1].split()[-1]] + [' '.join(x.split()[2:]) for x in meta[2:-1]]\n",
      "        vals = [(numReads,100)] + [(int(x.split()[:2][0]),float(x.split()[:2][1][1:-2])) for x in meta[2:]]\n",
      "        kv = dict(zip(keys,vals))\n",
      "        trimmLib[prev] = kv\n",
      "\n",
      "labMeta = []\n",
      "for lib, meta in trimmLib.items():\n",
      "    labMeta.append((illum[lib], meta['Reads'][0], meta['Dropped'][1], meta['Surviving'][0]))\n",
      "    \n",
      "labMeta.sort(key = lambda x : x[0])\n",
      "#filter out \"Rajewsky asexual(SRR125365)\". It only contains 50 reads. Wrongly uploaded file?\n",
      "labMeta = filter(lambda x : x[0][0] != 'SRR125365', labMeta)\n",
      "\n",
      "print 'Total reads (pairs):', commas(sum([x[1] for x in labMeta]))\n",
      "print 'Asexual reads (pairs):', commas(sum([x[1] for x in labMeta if x[0][3] == 'asexual']))\n",
      "print 'Sexual reads (pairs):', commas(sum([x[1] for x in labMeta if x[0][3] == 'sexual']))\n",
      "print\n",
      "print 'Total reads (pairs) after trimming:', commas(sum([x[3] for x in labMeta]))\n",
      "print 'Asexual reads (pairs) after trimming:', commas(sum([x[3] for x in labMeta if x[0][3] == 'asexual']))\n",
      "print 'Sexual reads (pairs) after trimming:', commas(sum([x[3] for x in labMeta if x[0][3] == 'sexual']))\n",
      "print\n",
      "for lab in list(set([x[0][2] for x in labMeta])):\n",
      "    print 'Total ' + lab + ' reads (pairs):', commas(sum([x[3] for x in labMeta if x[0][2] == lab]))\n",
      "trimmTable = ListTable()\n",
      "trimmTable.append(['Lab','strain','Run','Platform','Total Reads (Pairs)','Surviving','Percentage Dropped','Study'])\n",
      "trimmTable.extend([[x[0][2], x[0][3], x[0][0], x[0][9], commas(x[1]), commas(x[3]), str(x[2]),x[0][10]] for x in labMeta])\n",
      "trimmTable"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Total reads (pairs): 2,415,408,526\n",
        "Asexual reads (pairs): 1,755,332,176\n",
        "Sexual reads (pairs): 294,637,702\n",
        "\n",
        "Total reads (pairs) after trimming: 2,255,403,529\n",
        "Asexual reads (pairs) after trimming: 1,644,772,798\n",
        "Sexual reads (pairs) after trimming: 247,897,411\n",
        "\n",
        "Total Sanchez reads (pairs): 275,213,221\n",
        "Total Rajewsky reads (pairs): 125,853,711\n",
        "Total Pearson reads (pairs): 572,175,392\n",
        "Total Bartscherer reads (pairs): 422,055,448\n",
        "Total Graveley reads (pairs): 462,246,186\n",
        "Total Newmark reads (pairs): 160,942,119\n",
        "Total Rink reads (pairs): 236,917,452\n"
       ]
      },
      {
       "html": [
        "<table style='font-size:12px'><tr style='font-weight:bold'><td style='padding:4px 15px 4px 15px;border-color:white;text-align:center'>Lab</td><td style='padding:4px 15px 4px 15px;border-color:white;text-align:center'>strain</td><td style='padding:4px 15px 4px 15px;border-color:white;text-align:center'>Run</td><td style='padding:4px 15px 4px 15px;border-color:white;text-align:center'>Platform</td><td style='padding:4px 15px 4px 15px;border-color:white;text-align:center'>Total Reads (Pairs)</td><td style='padding:4px 15px 4px 15px;border-color:white;text-align:center'>Surviving</td><td style='padding:4px 15px 4px 15px;border-color:white;text-align:center'>Percentage Dropped</td><td style='padding:4px 15px 4px 15px;border-color:white;text-align:center'>Study</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>ERR032066</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>24,250,265</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>21,908,242</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.79</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>The head-regeneration transcriptome of the planarian Schmidtea mediterranea</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>ERR032067</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>24,874,216</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>22,508,798</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.79</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>The head-regeneration transcriptome of the planarian Schmidtea mediterranea</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>ERR032068</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>20,600,012</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>18,490,822</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.67</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>The head-regeneration transcriptome of the planarian Schmidtea mediterranea</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>ERR032069</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>24,298,493</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>21,953,262</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.83</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>The head-regeneration transcriptome of the planarian Schmidtea mediterranea</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>ERR032070</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>28,837,238</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>24,980,312</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>1.81</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>The head-regeneration transcriptome of the planarian Schmidtea mediterranea</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>ERR032071</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>23,720,712</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>21,032,476</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>1.11</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>The head-regeneration transcriptome of the planarian Schmidtea mediterranea</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>ERR032072</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>21,618,743</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>19,510,878</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.8</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>The head-regeneration transcriptome of the planarian Schmidtea mediterranea</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR1012785</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiScanSQ</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>33,653,372</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>33,593,725</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.18</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>2012-May FoxD RNAi (0dpa) 1</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR1012831</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiScanSQ</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>49,135,234</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>49,038,031</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.2</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>2012-May FoxD RNAi (3dpa) 1</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR1012832</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiScanSQ</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>19,937,670</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>19,872,705</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.33</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>2012-May FoxD RNAi (3dpa) 2</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR1012833</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiScanSQ</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>52,774,578</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>52,718,195</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.11</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>2012-May GFP RNAi (0dpa) 1</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR1012834</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiScanSQ</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>26,924,637</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>26,889,925</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.13</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>2012-May GFP RNAi (0dpa) 2</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR1012835</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiScanSQ</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>40,377,751</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>40,327,978</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.12</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>2012-May GFP RNAi (3dpa) 1</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR1012836</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiScanSQ</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>26,989,161</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>26,896,402</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.34</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>2012-May GFP RNAi (3dpa) 2</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Bartscherer</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR1012838</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiScanSQ</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>22,361,800</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>22,333,697</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.13</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>2012-May FoxD RNAi (0dpa) 2</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rajewsky</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR125366</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>20,259,434</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>19,371,400</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>4.38</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Single-end cDNA sequencing using Illumina GAIIX</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR388701</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>5,285,757</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>4,474,507</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>15.35</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM847461: Asexual Irradiated; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR388703</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>4,721,228</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>4,025,815</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>14.73</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM847462: Asexual Non-irradiated [single end]; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR388705</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>2,811,997</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>2,689,495</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.69</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM847463: Asexual Non-irradiated [paired end]; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>sexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR388706</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>5,715,834</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>4,936,011</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>13.64</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM847464: Sexual Irradiated; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>sexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR388708</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>5,580,560</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>4,079,827</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>26.89</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM847465: Sexual Non-irradiated [single end]; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>sexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR388710</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>4,529,559</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>4,071,979</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>1.3</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM847466: Sexual Non-irradiated [paired end]; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>sexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR388711</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>25,471,067</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>25,046,849</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>1.67</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM847467: Sexual Non-irradiated, X1 FACS purified neoblast; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>sexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR388712</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>25,378,801</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>25,020,790</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>1.41</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM847468: Sexual Non-irradiated, X2 FACS purified neoblast; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>sexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR388713</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>25,543,826</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>25,167,593</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>1.47</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM847469: Sexual Non-irradiated, Xins FACS purified differentiated cells; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Newmark</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR390350</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>38,650,598</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>38,645,351</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.01</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Schmidtea mediterranea Control1 (for PRMT5)</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Newmark</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR390351</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>29,434,582</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>29,425,764</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.03</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Schmidtea mediterranea Control2 (for PRMT5)</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Newmark</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR390352</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>51,951,765</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>51,941,728</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.02</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Schmidtea mediterranea PRMT5(RNAi)-1</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Newmark</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR390353</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>40,958,472</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>40,929,276</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.07</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Schmidtea mediterranea PRMT5(RNAi)-1</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR496276</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>21,917,357</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>15,720,729</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.41</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM929778: Stem Cells; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR496278</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>103,349,233</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>78,960,371</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>7.72</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM929779: Stem Cell Progeny; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR496280</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>20,805,822</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>19,393,597</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.25</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM929780: Differentiated tissues; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>None</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR654077</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer IIx</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>8,150,516</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>6,159,424</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>24.43</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Identification of neoblast- and regeneration-specific miRNAs in the planarian Schmidtea mediterranea</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Graveley</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>None</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR654079</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 1000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>357,288,132</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>356,573,896</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.2</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Identification of neoblast- and regeneration-specific miRNAs in the planarian Schmidtea mediterranea</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR867380</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>44,120,681</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>43,720,481</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.91</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM1147717: lhx1/5-1(RNAi) animals 1; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR867381</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>41,070,364</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>40,809,618</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.63</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM1147718: lhx1/5-1(RNAi) animals 2; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR867382</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>46,498,174</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>46,389,747</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.23</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM1147719: lhx1/5-1(RNAi) animals 3; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR867383</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>60,414,361</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>60,266,322</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.25</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM1147720: pitx(RNAi) animals 1; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR867384</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>50,708,699</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>50,591,727</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.23</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM1147721: pitx(RNAi) animals 2; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR867385</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>44,472,617</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>44,181,512</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.65</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM1147722: pitx(RNAi) animals 3; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR867386</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>51,796,391</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>51,654,655</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.27</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM1147723: control(RNAi) animals (unc22) 1; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR867387</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>61,913,948</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>61,809,609</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.17</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM1147724: control(RNAi) animals (unc22) 2; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Pearson</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR867388</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>58,809,965</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>58,677,024</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.23</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>GSM1147725: control(RNAi) animals (unc22) 3; Schmidtea mediterranea; RNA-Seq</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rink</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR900846</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>23,801,898</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>23,752,864</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.21</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>single-end sequencing raw reads from regenerating Smed tail wounds at time point 0h</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rink</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR900847</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>29,381,516</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>29,326,285</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.19</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>single-end sequencing raw reads from regenerating Smed tail wounds at time point 4h post amputation</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rink</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR900848</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>30,743,506</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>30,690,948</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.17</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>single-end sequencing raw reads from regenerating Smed tail wounds at time point 12h post amputation</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rink</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR900849</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>29,896,037</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>29,846,804</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.16</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>single-end sequencing raw reads from regenerating Smed tail wounds at time point 16h post amputation</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rink</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR900850</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>25,247,322</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>25,193,694</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.21</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>single-end sequencing raw reads from regenerating Smed tail wounds at time point 24h post amputation</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rink</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR900851</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>32,948,999</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>32,883,126</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.2</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>single-end sequencing raw reads from regenerating Smed tail wounds at time point 48h post amputation</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rink</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR900852</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>31,539,544</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>31,485,717</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.17</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>single-end sequencing raw reads from regenerating Smed tail wounds at time point 72h post amputation</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rink</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR900853</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>33,804,456</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>33,738,014</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.2</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>single-end sequencing raw reads from regenerating Smed tail wounds at time point 120h post amputation</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Sanchez</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR955099</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>144,975,033</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>115,638,859</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.19</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>RNA sequencing for transcriptome assembly - Smed Asexual(CIW4) biotype</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Sanchez</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>sexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>SRR955511</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina HiSeq 2000</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>202,418,055</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>159,574,362</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>0.2</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>RNA sequencing for transcriptome assembly - Smed Sexual(S2F2) biotype</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rajewsky</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>oenals4</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>16,813,053</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>12,540,425</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>4.37</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>None</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rajewsky</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>oenals5</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>15,823,736</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>11,771,232</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>4.47</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>None</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rajewsky</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>oenalx1</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>32,100,429</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>26,241,570</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>1.01</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>None</td></tr><tr style='background:#E8E8E8'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rajewsky</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>oenalx2</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>34,363,267</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>28,094,218</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>1.32</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>None</td></tr><tr style='background:white'><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Rajewsky</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>asexual</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>oenalxins</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>Illumina Genome Analyzer II</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>33,588,053</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>27,834,866</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>1.47</td><td style='padding:4px 15px 4px 15px;border-color:#B0B0B0;text-align:center'>None</td></tr></table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 3,
       "text": [
        "[['Lab',\n",
        "  'strain',\n",
        "  'Run',\n",
        "  'Platform',\n",
        "  'Total Reads (Pairs)',\n",
        "  'Surviving',\n",
        "  'Percentage Dropped',\n",
        "  'Study'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'ERR032066',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '24,250,265',\n",
        "  '21,908,242',\n",
        "  '0.79',\n",
        "  'The head-regeneration transcriptome of the planarian Schmidtea mediterranea'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'ERR032067',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '24,874,216',\n",
        "  '22,508,798',\n",
        "  '0.79',\n",
        "  'The head-regeneration transcriptome of the planarian Schmidtea mediterranea'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'ERR032068',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '20,600,012',\n",
        "  '18,490,822',\n",
        "  '0.67',\n",
        "  'The head-regeneration transcriptome of the planarian Schmidtea mediterranea'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'ERR032069',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '24,298,493',\n",
        "  '21,953,262',\n",
        "  '0.83',\n",
        "  'The head-regeneration transcriptome of the planarian Schmidtea mediterranea'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'ERR032070',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '28,837,238',\n",
        "  '24,980,312',\n",
        "  '1.81',\n",
        "  'The head-regeneration transcriptome of the planarian Schmidtea mediterranea'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'ERR032071',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '23,720,712',\n",
        "  '21,032,476',\n",
        "  '1.11',\n",
        "  'The head-regeneration transcriptome of the planarian Schmidtea mediterranea'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'ERR032072',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '21,618,743',\n",
        "  '19,510,878',\n",
        "  '0.8',\n",
        "  'The head-regeneration transcriptome of the planarian Schmidtea mediterranea'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'SRR1012785',\n",
        "  'Illumina HiScanSQ',\n",
        "  '33,653,372',\n",
        "  '33,593,725',\n",
        "  '0.18',\n",
        "  '2012-May FoxD RNAi (0dpa) 1'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'SRR1012831',\n",
        "  'Illumina HiScanSQ',\n",
        "  '49,135,234',\n",
        "  '49,038,031',\n",
        "  '0.2',\n",
        "  '2012-May FoxD RNAi (3dpa) 1'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'SRR1012832',\n",
        "  'Illumina HiScanSQ',\n",
        "  '19,937,670',\n",
        "  '19,872,705',\n",
        "  '0.33',\n",
        "  '2012-May FoxD RNAi (3dpa) 2'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'SRR1012833',\n",
        "  'Illumina HiScanSQ',\n",
        "  '52,774,578',\n",
        "  '52,718,195',\n",
        "  '0.11',\n",
        "  '2012-May GFP RNAi (0dpa) 1'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'SRR1012834',\n",
        "  'Illumina HiScanSQ',\n",
        "  '26,924,637',\n",
        "  '26,889,925',\n",
        "  '0.13',\n",
        "  '2012-May GFP RNAi (0dpa) 2'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'SRR1012835',\n",
        "  'Illumina HiScanSQ',\n",
        "  '40,377,751',\n",
        "  '40,327,978',\n",
        "  '0.12',\n",
        "  '2012-May GFP RNAi (3dpa) 1'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'SRR1012836',\n",
        "  'Illumina HiScanSQ',\n",
        "  '26,989,161',\n",
        "  '26,896,402',\n",
        "  '0.34',\n",
        "  '2012-May GFP RNAi (3dpa) 2'],\n",
        " ['Bartscherer',\n",
        "  'asexual',\n",
        "  'SRR1012838',\n",
        "  'Illumina HiScanSQ',\n",
        "  '22,361,800',\n",
        "  '22,333,697',\n",
        "  '0.13',\n",
        "  '2012-May FoxD RNAi (0dpa) 2'],\n",
        " ['Rajewsky',\n",
        "  'asexual',\n",
        "  'SRR125366',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '20,259,434',\n",
        "  '19,371,400',\n",
        "  '4.38',\n",
        "  'Single-end cDNA sequencing using Illumina GAIIX'],\n",
        " ['Graveley',\n",
        "  'asexual',\n",
        "  'SRR388701',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '5,285,757',\n",
        "  '4,474,507',\n",
        "  '15.35',\n",
        "  'GSM847461: Asexual Irradiated; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Graveley',\n",
        "  'asexual',\n",
        "  'SRR388703',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '4,721,228',\n",
        "  '4,025,815',\n",
        "  '14.73',\n",
        "  'GSM847462: Asexual Non-irradiated [single end]; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Graveley',\n",
        "  'asexual',\n",
        "  'SRR388705',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '2,811,997',\n",
        "  '2,689,495',\n",
        "  '0.69',\n",
        "  'GSM847463: Asexual Non-irradiated [paired end]; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Graveley',\n",
        "  'sexual',\n",
        "  'SRR388706',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '5,715,834',\n",
        "  '4,936,011',\n",
        "  '13.64',\n",
        "  'GSM847464: Sexual Irradiated; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Graveley',\n",
        "  'sexual',\n",
        "  'SRR388708',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '5,580,560',\n",
        "  '4,079,827',\n",
        "  '26.89',\n",
        "  'GSM847465: Sexual Non-irradiated [single end]; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Graveley',\n",
        "  'sexual',\n",
        "  'SRR388710',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '4,529,559',\n",
        "  '4,071,979',\n",
        "  '1.3',\n",
        "  'GSM847466: Sexual Non-irradiated [paired end]; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Graveley',\n",
        "  'sexual',\n",
        "  'SRR388711',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '25,471,067',\n",
        "  '25,046,849',\n",
        "  '1.67',\n",
        "  'GSM847467: Sexual Non-irradiated, X1 FACS purified neoblast; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Graveley',\n",
        "  'sexual',\n",
        "  'SRR388712',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '25,378,801',\n",
        "  '25,020,790',\n",
        "  '1.41',\n",
        "  'GSM847468: Sexual Non-irradiated, X2 FACS purified neoblast; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Graveley',\n",
        "  'sexual',\n",
        "  'SRR388713',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '25,543,826',\n",
        "  '25,167,593',\n",
        "  '1.47',\n",
        "  'GSM847469: Sexual Non-irradiated, Xins FACS purified differentiated cells; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Newmark',\n",
        "  'asexual',\n",
        "  'SRR390350',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '38,650,598',\n",
        "  '38,645,351',\n",
        "  '0.01',\n",
        "  'Schmidtea mediterranea Control1 (for PRMT5)'],\n",
        " ['Newmark',\n",
        "  'asexual',\n",
        "  'SRR390351',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '29,434,582',\n",
        "  '29,425,764',\n",
        "  '0.03',\n",
        "  'Schmidtea mediterranea Control2 (for PRMT5)'],\n",
        " ['Newmark',\n",
        "  'asexual',\n",
        "  'SRR390352',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '51,951,765',\n",
        "  '51,941,728',\n",
        "  '0.02',\n",
        "  'Schmidtea mediterranea PRMT5(RNAi)-1'],\n",
        " ['Newmark',\n",
        "  'asexual',\n",
        "  'SRR390353',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '40,958,472',\n",
        "  '40,929,276',\n",
        "  '0.07',\n",
        "  'Schmidtea mediterranea PRMT5(RNAi)-1'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR496276',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '21,917,357',\n",
        "  '15,720,729',\n",
        "  '0.41',\n",
        "  'GSM929778: Stem Cells; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR496278',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '103,349,233',\n",
        "  '78,960,371',\n",
        "  '7.72',\n",
        "  'GSM929779: Stem Cell Progeny; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR496280',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '20,805,822',\n",
        "  '19,393,597',\n",
        "  '0.25',\n",
        "  'GSM929780: Differentiated tissues; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Graveley',\n",
        "  'None',\n",
        "  'SRR654077',\n",
        "  'Illumina Genome Analyzer IIx',\n",
        "  '8,150,516',\n",
        "  '6,159,424',\n",
        "  '24.43',\n",
        "  'Identification of neoblast- and regeneration-specific miRNAs in the planarian Schmidtea mediterranea'],\n",
        " ['Graveley',\n",
        "  'None',\n",
        "  'SRR654079',\n",
        "  'Illumina HiSeq 1000',\n",
        "  '357,288,132',\n",
        "  '356,573,896',\n",
        "  '0.2',\n",
        "  'Identification of neoblast- and regeneration-specific miRNAs in the planarian Schmidtea mediterranea'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR867380',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '44,120,681',\n",
        "  '43,720,481',\n",
        "  '0.91',\n",
        "  'GSM1147717: lhx1/5-1(RNAi) animals 1; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR867381',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '41,070,364',\n",
        "  '40,809,618',\n",
        "  '0.63',\n",
        "  'GSM1147718: lhx1/5-1(RNAi) animals 2; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR867382',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '46,498,174',\n",
        "  '46,389,747',\n",
        "  '0.23',\n",
        "  'GSM1147719: lhx1/5-1(RNAi) animals 3; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR867383',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '60,414,361',\n",
        "  '60,266,322',\n",
        "  '0.25',\n",
        "  'GSM1147720: pitx(RNAi) animals 1; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR867384',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '50,708,699',\n",
        "  '50,591,727',\n",
        "  '0.23',\n",
        "  'GSM1147721: pitx(RNAi) animals 2; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR867385',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '44,472,617',\n",
        "  '44,181,512',\n",
        "  '0.65',\n",
        "  'GSM1147722: pitx(RNAi) animals 3; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR867386',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '51,796,391',\n",
        "  '51,654,655',\n",
        "  '0.27',\n",
        "  'GSM1147723: control(RNAi) animals (unc22) 1; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR867387',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '61,913,948',\n",
        "  '61,809,609',\n",
        "  '0.17',\n",
        "  'GSM1147724: control(RNAi) animals (unc22) 2; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Pearson',\n",
        "  'asexual',\n",
        "  'SRR867388',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '58,809,965',\n",
        "  '58,677,024',\n",
        "  '0.23',\n",
        "  'GSM1147725: control(RNAi) animals (unc22) 3; Schmidtea mediterranea; RNA-Seq'],\n",
        " ['Rink',\n",
        "  'asexual',\n",
        "  'SRR900846',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '23,801,898',\n",
        "  '23,752,864',\n",
        "  '0.21',\n",
        "  'single-end sequencing raw reads from regenerating Smed tail wounds at time point 0h'],\n",
        " ['Rink',\n",
        "  'asexual',\n",
        "  'SRR900847',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '29,381,516',\n",
        "  '29,326,285',\n",
        "  '0.19',\n",
        "  'single-end sequencing raw reads from regenerating Smed tail wounds at time point 4h post amputation'],\n",
        " ['Rink',\n",
        "  'asexual',\n",
        "  'SRR900848',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '30,743,506',\n",
        "  '30,690,948',\n",
        "  '0.17',\n",
        "  'single-end sequencing raw reads from regenerating Smed tail wounds at time point 12h post amputation'],\n",
        " ['Rink',\n",
        "  'asexual',\n",
        "  'SRR900849',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '29,896,037',\n",
        "  '29,846,804',\n",
        "  '0.16',\n",
        "  'single-end sequencing raw reads from regenerating Smed tail wounds at time point 16h post amputation'],\n",
        " ['Rink',\n",
        "  'asexual',\n",
        "  'SRR900850',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '25,247,322',\n",
        "  '25,193,694',\n",
        "  '0.21',\n",
        "  'single-end sequencing raw reads from regenerating Smed tail wounds at time point 24h post amputation'],\n",
        " ['Rink',\n",
        "  'asexual',\n",
        "  'SRR900851',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '32,948,999',\n",
        "  '32,883,126',\n",
        "  '0.2',\n",
        "  'single-end sequencing raw reads from regenerating Smed tail wounds at time point 48h post amputation'],\n",
        " ['Rink',\n",
        "  'asexual',\n",
        "  'SRR900852',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '31,539,544',\n",
        "  '31,485,717',\n",
        "  '0.17',\n",
        "  'single-end sequencing raw reads from regenerating Smed tail wounds at time point 72h post amputation'],\n",
        " ['Rink',\n",
        "  'asexual',\n",
        "  'SRR900853',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '33,804,456',\n",
        "  '33,738,014',\n",
        "  '0.2',\n",
        "  'single-end sequencing raw reads from regenerating Smed tail wounds at time point 120h post amputation'],\n",
        " ['Sanchez',\n",
        "  'asexual',\n",
        "  'SRR955099',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '144,975,033',\n",
        "  '115,638,859',\n",
        "  '0.19',\n",
        "  'RNA sequencing for transcriptome assembly - Smed Asexual(CIW4) biotype'],\n",
        " ['Sanchez',\n",
        "  'sexual',\n",
        "  'SRR955511',\n",
        "  'Illumina HiSeq 2000',\n",
        "  '202,418,055',\n",
        "  '159,574,362',\n",
        "  '0.2',\n",
        "  'RNA sequencing for transcriptome assembly - Smed Sexual(S2F2) biotype'],\n",
        " ['Rajewsky',\n",
        "  'asexual',\n",
        "  'oenals4',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '16,813,053',\n",
        "  '12,540,425',\n",
        "  '4.37',\n",
        "  'None'],\n",
        " ['Rajewsky',\n",
        "  'asexual',\n",
        "  'oenals5',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '15,823,736',\n",
        "  '11,771,232',\n",
        "  '4.47',\n",
        "  'None'],\n",
        " ['Rajewsky',\n",
        "  'asexual',\n",
        "  'oenalx1',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '32,100,429',\n",
        "  '26,241,570',\n",
        "  '1.01',\n",
        "  'None'],\n",
        " ['Rajewsky',\n",
        "  'asexual',\n",
        "  'oenalx2',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '34,363,267',\n",
        "  '28,094,218',\n",
        "  '1.32',\n",
        "  'None'],\n",
        " ['Rajewsky',\n",
        "  'asexual',\n",
        "  'oenalxins',\n",
        "  'Illumina Genome Analyzer II',\n",
        "  '33,588,053',\n",
        "  '27,834,866',\n",
        "  '1.47',\n",
        "  'None']]"
       ]
      }
     ],
     "prompt_number": 3
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### Map reads with BWA\n",
      "\n",
      "Trimmed PE and SE are mapped to the genome. Trimmed unpaired reads will be mapped as SE reads. "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "trimmedFs = [x for x in os.listdir('/home/share/projects/smed_strain/trimm') if x.split('.')[-1] == 'gz']\n",
      "print 'Number of .fastq files:', len(trimmedFs)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Number of .fastq files: 119\n"
       ]
      }
     ],
     "prompt_number": 4
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "The paired fastq files don't have the corrected formatted fastq headers. Wrote an interleave script to rename header and pipe to bwa as an interleaved file."
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#Interleave script\n",
      "HTML(textarea('Interleave python script', '''import sys,gzip\n",
      "\n",
      "pe1 = gzip.open(sys.argv[1],'rb')\n",
      "pe2 = gzip.open(sys.argv[2],'rb')\n",
      "\n",
      "for line in pe1:\n",
      "        print '.'.join(line.split()[0].split('.')[:-1]) + \"\\\\n\" + \\\\\n",
      "        pe1.next() + \\\\\n",
      "        '.'.join(pe1.next().split()[0].split('.')[:-1]) + '\\\\n' + \\\\\n",
      "        pe1.next() + \\\\\n",
      "        '.'.join(pe2.next().split()[0].split('.')[:-1]) + '\\\\n' + \\\\\n",
      "        pe2.next() + \\\\\n",
      "        '.'.join(pe2.next().split()[0].split('.')[:-1]) + '\\\\n' + \\\\\n",
      "        pe2.next().strip()\n",
      "'''))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<b>Interleave python script</b>:<br>\n",
        "<textarea style=\"width: 1000px; height:200px\">\n",
        "import sys,gzip\n",
        "\n",
        "pe1 = gzip.open(sys.argv[1],'rb')\n",
        "pe2 = gzip.open(sys.argv[2],'rb')\n",
        "\n",
        "for line in pe1:\n",
        "        print '.'.join(line.split()[0].split('.')[:-1]) + \"\\n\" + \\\n",
        "        pe1.next() + \\\n",
        "        '.'.join(pe1.next().split()[0].split('.')[:-1]) + '\\n' + \\\n",
        "        pe1.next() + \\\n",
        "        '.'.join(pe2.next().split()[0].split('.')[:-1]) + '\\n' + \\\n",
        "        pe2.next() + \\\n",
        "        '.'.join(pe2.next().split()[0].split('.')[:-1]) + '\\n' + \\\n",
        "        pe2.next().strip()\n",
        "\n",
        "</textarea>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 5,
       "text": [
        "<IPython.core.display.HTML at 0x1b86650>"
       ]
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "prePath = '/home/share/projects/smed_strain/trimm/'\n",
      "pecmdA = 'python interleave.py '  \n",
      "pecmdB = ' | bwa mem -t 32 -p -R \"'\n",
      "pecmdC = '\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - ' \n",
      "\n",
      "secmdA = 'bwa mem -t 32 -R \"'\n",
      "secmdB = '\" /home/share/genome/smed/bwa/smed3.1 '\n",
      "secmdC = ' | samtools view -hSbu - | samtools sort - '\n",
      "\n",
      "oenalCmdA = 'bwa mem -t 32 -R \"'\n",
      "oenalCmdB = '\" /home/share/genome/smed/bwa/smed3.1 '\n",
      "oenalCmdC = ' | samtools view -hSbu - | samtools sort - '\n",
      "\n",
      "mappingCmds = ''\n",
      "for item in labMeta:\n",
      "    run = item[0][0]\n",
      "    fastqs = [x for x in trimmedFs if x.split('_')[0] == run]\n",
      "    rg = '@RG\\\\tID:' + run + '\\\\tlab:' + item[0][2] + '\\\\tstrain:' + item[0][3]\n",
      "    if run.find('oenal') == -1:\n",
      "        if len(fastqs) == 1:\n",
      "            #SE \n",
      "            se = prePath + fastqs[0]\n",
      "            mappingCmds += secmdA + rg + secmdB + se + secmdC + run + '.se.sorted\\n'\n",
      "        else:\n",
      "            #PE\n",
      "            pe1 = prePath + run + '_1.paired.trimmed.fastq.gz'\n",
      "            pe2 = prePath + run + '_2.paired.trimmed.fastq.gz'\n",
      "            mappingCmds += pecmdA + pe1 + ' ' + pe2 + pecmdB + rg + pecmdC + run + '.pe.sorted\\n'\n",
      "\n",
      "            #map unpaired as SE\n",
      "            se1 = prePath + run + '_1.unpaired.trimmed.fastq.gz'\n",
      "            se2 = prePath + run + '_2.unpaired.trimmed.fastq.gz'\n",
      "            mappingCmds += secmdA + rg + secmdB + se1 + secmdC + run + '.unpe1.sorted\\n'\n",
      "            mappingCmds += secmdA + rg + secmdB + se2 + secmdC + run + '.unpe2.sorted\\n'\n",
      "    else:\n",
      "        if len(fastqs) == 1:\n",
      "            #SE \n",
      "            se = prePath + fastqs[0]\n",
      "            mappingCmds += secmdA + rg + secmdB + se + secmdC + run + '.se.sorted\\n'\n",
      "        else:\n",
      "            #PE\n",
      "            pe1 = prePath + run + '_1.paired.trimmed.fastq.gz'\n",
      "            pe2 = prePath + run + '_2.paired.trimmed.fastq.gz'\n",
      "            mappingCmds += oenalCmdA + rg + oenalCmdB + pe1 + ' ' + pe2 + oenalCmdC + run + '.pe.sorted\\n'\n",
      "\n",
      "            #map unpaired as SE\n",
      "            se1 = prePath + run + '_1.unpaired.trimmed.fastq.gz'\n",
      "            se2 = prePath + run + '_2.unpaired.trimmed.fastq.gz'\n",
      "            mappingCmds +=  secmdA + rg + secmdB + se1 + secmdC + run + '.unpe1.sorted\\n'\n",
      "            mappingCmds +=  secmdA + rg + secmdB + se2 + secmdC + run + '.unpe2.sorted\\n'\n",
      "            \n",
      "HTML(textarea('BWA Mapping bash commands', mappingCmds))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<b>BWA Mapping bash commands</b>:<br>\n",
        "<textarea style=\"width: 1000px; height:200px\">\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/ERR032066_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/ERR032066_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:ERR032066\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - ERR032066.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032066\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032066_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032066.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032066\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032066_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032066.unpe2.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/ERR032067_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/ERR032067_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:ERR032067\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - ERR032067.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032067\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032067_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032067.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032067\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032067_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032067.unpe2.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/ERR032068_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/ERR032068_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:ERR032068\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - ERR032068.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032068\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032068_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032068.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032068\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032068_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032068.unpe2.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/ERR032069_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/ERR032069_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:ERR032069\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - ERR032069.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032069\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032069_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032069.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032069\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032069_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032069.unpe2.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/ERR032070_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/ERR032070_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:ERR032070\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - ERR032070.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032070\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032070_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032070.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032070\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032070_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032070.unpe2.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/ERR032071_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/ERR032071_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:ERR032071\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - ERR032071.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032071\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032071_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032071.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032071\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032071_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032071.unpe2.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/ERR032072_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/ERR032072_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:ERR032072\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - ERR032072.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032072\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032072_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032072.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:ERR032072\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/ERR032072_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - ERR032072.unpe2.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR1012785\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR1012785_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR1012785.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR1012831\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR1012831_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR1012831.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR1012832\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR1012832_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR1012832.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR1012833\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR1012833_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR1012833.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR1012834\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR1012834_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR1012834.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR1012835\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR1012835_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR1012835.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR1012836\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR1012836_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR1012836.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR1012838\\tlab:Bartscherer\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR1012838_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR1012838.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR125366\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR125366_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR125366.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388701\\tlab:Graveley\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388701_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388701.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388703\\tlab:Graveley\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388703_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388703.se.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/SRR388705_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/SRR388705_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:SRR388705\\tlab:Graveley\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - SRR388705.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388705\\tlab:Graveley\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388705_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388705.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388705\\tlab:Graveley\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388705_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388705.unpe2.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388706\\tlab:Graveley\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388706_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388706.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388708\\tlab:Graveley\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388708_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388708.se.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/SRR388710_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/SRR388710_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:SRR388710\\tlab:Graveley\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - SRR388710.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388710\\tlab:Graveley\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388710_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388710.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388710\\tlab:Graveley\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388710_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388710.unpe2.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388711\\tlab:Graveley\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388711_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388711.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388712\\tlab:Graveley\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388712_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388712.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR388713\\tlab:Graveley\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR388713_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR388713.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR390350\\tlab:Newmark\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR390350_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR390350.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR390351\\tlab:Newmark\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR390351_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR390351.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR390352\\tlab:Newmark\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR390352_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR390352.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR390353\\tlab:Newmark\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR390353_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR390353.se.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/SRR496276_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/SRR496276_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:SRR496276\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - SRR496276.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR496276\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR496276_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR496276.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR496276\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR496276_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR496276.unpe2.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/SRR496278_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/SRR496278_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:SRR496278\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - SRR496278.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR496278\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR496278_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR496278.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR496278\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR496278_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR496278.unpe2.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/SRR496280_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/SRR496280_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:SRR496280\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - SRR496280.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR496280\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR496280_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR496280.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR496280\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR496280_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR496280.unpe2.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR654077\\tlab:Graveley\\tstrain:None\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR654077_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR654077.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR654079\\tlab:Graveley\\tstrain:None\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR654079_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR654079.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR867380\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR867380_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR867380.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR867381\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR867381_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR867381.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR867382\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR867382_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR867382.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR867383\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR867383_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR867383.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR867384\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR867384_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR867384.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR867385\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR867385_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR867385.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR867386\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR867386_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR867386.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR867387\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR867387_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR867387.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR867388\\tlab:Pearson\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR867388_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR867388.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR900846\\tlab:Rink\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR900846_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR900846.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR900847\\tlab:Rink\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR900847_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR900847.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR900848\\tlab:Rink\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR900848_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR900848.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR900849\\tlab:Rink\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR900849_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR900849.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR900850\\tlab:Rink\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR900850_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR900850.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR900851\\tlab:Rink\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR900851_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR900851.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR900852\\tlab:Rink\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR900852_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR900852.se.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR900853\\tlab:Rink\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR900853_1.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR900853.se.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/SRR955099_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/SRR955099_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:SRR955099\\tlab:Sanchez\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - SRR955099.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR955099\\tlab:Sanchez\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR955099_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR955099.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR955099\\tlab:Sanchez\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR955099_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR955099.unpe2.sorted\n",
        "python interleave.py /home/share/projects/smed_strain/trimm/SRR955511_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/SRR955511_2.paired.trimmed.fastq.gz | bwa mem -t 32 -p -R \"@RG\\tID:SRR955511\\tlab:Sanchez\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 - | samtools view -hSbu - | samtools sort - SRR955511.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR955511\\tlab:Sanchez\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR955511_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR955511.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:SRR955511\\tlab:Sanchez\\tstrain:sexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/SRR955511_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - SRR955511.unpe2.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenals4\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenals4_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/oenals4_2.paired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenals4.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenals4\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenals4_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenals4.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenals4\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenals4_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenals4.unpe2.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenals5\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenals5_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/oenals5_2.paired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenals5.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenals5\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenals5_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenals5.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenals5\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenals5_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenals5.unpe2.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenalx1\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenalx1_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/oenalx1_2.paired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenalx1.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenalx1\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenalx1_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenalx1.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenalx1\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenalx1_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenalx1.unpe2.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenalx2\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenalx2_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/oenalx2_2.paired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenalx2.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenalx2\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenalx2_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenalx2.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenalx2\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenalx2_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenalx2.unpe2.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenalxins\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenalxins_1.paired.trimmed.fastq.gz /home/share/projects/smed_strain/trimm/oenalxins_2.paired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenalxins.pe.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenalxins\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenalxins_1.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenalxins.unpe1.sorted\n",
        "bwa mem -t 32 -R \"@RG\\tID:oenalxins\\tlab:Rajewsky\\tstrain:asexual\" /home/share/genome/smed/bwa/smed3.1 /home/share/projects/smed_strain/trimm/oenalxins_2.unpaired.trimmed.fastq.gz | samtools view -hSbu - | samtools sort - oenalxins.unpe2.sorted\n",
        "\n",
        "</textarea>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 6,
       "text": [
        "<IPython.core.display.HTML at 0x1b92350>"
       ]
      }
     ],
     "prompt_number": 6
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### Map SOLiD reads with NovoalignCS\n",
      "\n",
      "Decided to use NovoalignCS rather than LifeScope because I don't have access to LifeScope anymore. \n",
      "\n",
      "XSQ_tools was used to extract `csfasta` and `_QV.qual` files from `.XSQ` files:\n",
      "\n",
      "    ./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc12_smed_gfp_t_7d_B.xsq\n",
      "\n",
      "    parallel :::: convert.bash\n",
      "\n",
      "Novoalign CS 1.05.00 was used to align each library:\n",
      "\n",
      "    novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f ../bc1_smed_reg_h_06h.csfasta -o SAM -F CSFASTAnQV -K ../bc1_smed_reg_h_06h.novoalign.mismatch"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#Map all regenerating timepoints, GFP RNAi controls or homeostasis libraries\n",
      "xsqs = [x for x in os.listdir('/home/share/ngsData/smed/solid/') if x.find('reg') != -1 or x.find('gfp') != -1 or x.find('hom') != -1]\n",
      "print \"Number of SOLiD libraries:\", len(xsqs)\n",
      "print\n",
      "print \"Libraries:\\n\", '\\n'.join(['   ' + x for x in xsqs])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Number of SOLiD libraries: 48\n",
        "\n",
        "Libraries:\n",
        "   bc12_smed_gfp_t_7d_B.xsq\n",
        "   bc9_smed_gfp_w_12h_F3.xsq\n",
        "   bc12_smed_gfp_t_7d_A.xsq\n",
        "   bc1_smed_reg_t_72h.xsq\n",
        "   bc2_smed_reg_t_24h.xsq\n",
        "   bc9_smed_hom_w_00h_F3.xsq\n",
        "   bc2_smed_reg_h_72h.xsq\n",
        "   bc9_smed_gfp_w_48h_F5.xsq\n",
        "   bc2_smed_reg_t_12h.xsq\n",
        "   bc2_smed_reg_h_12h.xsq\n",
        "   bc7_smed_reg_h_00h_A.xsq\n",
        "   bc8_smed_gfp_w_5d.xsq\n",
        "   bc2_smed_reg_t_36h.xsq\n",
        "   bc2_smed_reg_h_48h.xsq\n",
        "   bc7_smed_reg_t_36h.xsq\n",
        "   bc1_smed_reg_t_12h.xsq\n",
        "   bc7_smed_reg_t_48h.xsq\n",
        "   bc7_smed_reg_t_00h_B.xsq\n",
        "   bc1_smed_reg_t_24h.xsq\n",
        "   bc2_smed_reg_t_48h.xsq\n",
        "   bc1_smed_reg_t_06h.xsq\n",
        "   bc10_smed_gfp_20d_smgControl.xsq\n",
        "   bc1_smed_reg_t_36h.xsq\n",
        "   bc7_smed_reg_t_00h_A.xsq\n",
        "   bc10_smed_gfp_7d.xsq\n",
        "   bc2_smed_reg_t_72h.xsq\n",
        "   bc10_smed_gfp_18h.xsq\n",
        "   bc1_smed_reg_h_72h.xsq\n",
        "   bc1_smed_hom_w_00h.xsq\n",
        "   bc2_smed_reg_h_24h.xsq\n",
        "   bc9_smed_gfp_w_24h_F5.xsq\n",
        "   bc7_smed_reg_h_00h_B.xsq\n",
        "   bc10_smed_gfp_06h.xsq\n",
        "   bc2_smed_reg_h_06h.xsq\n",
        "   bc2_smed_hom_w_00h.xsq\n",
        "   bc1_smed_reg_t_48h.xsq\n",
        "   bc1_smed_reg_h_48h.xsq\n",
        "   bc9_smed_gfp_w_12h_F5.xsq\n",
        "   bc10_smed_gfp_20d.xsq\n",
        "   bc2_smed_reg_t_06h.xsq\n",
        "   bc1_smed_reg_h_06h.xsq\n",
        "   bc9_smed_hom_w_00h_F5.xsq\n",
        "   bc2_smed_reg_h_36h.xsq\n",
        "   bc1_smed_reg_h_24h.xsq\n",
        "   bc9_smed_gfp_w_48h_F3.xsq\n",
        "   bc1_smed_reg_h_36h.xsq\n",
        "   bc1_smed_reg_h_12h.xsq\n",
        "   bc9_smed_gfp_w_24h_F3.xsq\n"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#generate commands for converting to csfasta and qual files\n",
      "solidConvertCmds = ''\n",
      "for xsq in xsqs:\n",
      "    cmd = './convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/' + xsq\n",
      "    solidConvertCmds += cmd + '\\n'\n",
      "\n",
      "#rename csfasta/qual files\n",
      "\n",
      "'''\n",
      "csPaths = os.popen('ls /home/share/ngsData/smed/Libraries/*/*/reads/*.csfasta').read().strip().split('\\n')\n",
      "csPaths = [x.split('.')[0] for x in csPaths]\n",
      "solidRenameCmds = ''\n",
      "for csPath in csPaths:\n",
      "    meta = csPath.split('/')[-1].split('.')[0].split('_')\n",
      "    meta = meta[len(meta) / 2:]\n",
      "    if meta[-2] == 'F3' or meta[-2] == 'F5':\n",
      "        meta = meta[:-1]\n",
      "    name = '_'.join(meta)\n",
      "    \n",
      "    solidRenameCmds += 'mv ' + csPath + '.csfasta ' + name + '.csfasta\\n'\n",
      "    solidRenameCmds += 'mv ' + csPath + '.QV.qual ' + name + \"_QV.qual\\n\"\n",
      "  \n",
      "HTML(textarea('csfasta/qual rename bash commands',temp))\n",
      "'''\n",
      "\n",
      "HTML(textarea('XSQ conversion bash commands',solidConvertCmds))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<b>XSQ conversion bash commands</b>:<br>\n",
        "<textarea style=\"width: 1000px; height:200px\">\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc12_smed_gfp_t_7d_B.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc9_smed_gfp_w_12h_F3.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc12_smed_gfp_t_7d_A.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_t_72h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_t_24h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc9_smed_hom_w_00h_F3.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_h_72h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc9_smed_gfp_w_48h_F5.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_t_12h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_h_12h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc7_smed_reg_h_00h_A.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc8_smed_gfp_w_5d.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_t_36h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_h_48h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc7_smed_reg_t_36h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_t_12h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc7_smed_reg_t_48h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc7_smed_reg_t_00h_B.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_t_24h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_t_48h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_t_06h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc10_smed_gfp_20d_smgControl.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_t_36h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc7_smed_reg_t_00h_A.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc10_smed_gfp_7d.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_t_72h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc10_smed_gfp_18h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_h_72h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_hom_w_00h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_h_24h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc9_smed_gfp_w_24h_F5.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc7_smed_reg_h_00h_B.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc10_smed_gfp_06h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_h_06h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_hom_w_00h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_t_48h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_h_48h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc9_smed_gfp_w_12h_F5.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc10_smed_gfp_20d.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_t_06h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_h_06h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc9_smed_hom_w_00h_F5.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc2_smed_reg_h_36h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_h_24h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc9_smed_gfp_w_48h_F3.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_h_36h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc1_smed_reg_h_12h.xsq\n",
        "./convertFromXSQ.sh -c -o /home/share/ngsData/smed/ /home/share/ngsData/smed/solid/bc9_smed_gfp_w_24h_F3.xsq\n",
        "\n",
        "</textarea>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 8,
       "text": [
        "<IPython.core.display.HTML at 0x1b88f50>"
       ]
      }
     ],
     "prompt_number": 8
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#generate novoaligncs commands\n",
      "csFiles = ['/home/share/ngsData/smed/raw/' + x.split('.')[0] for x in os.listdir('/home/share/ngsData/smed/raw') if x.split('.')[-1] == 'csfasta']\n",
      "solidMapCmds = ''\n",
      "for csFile in csFiles:\n",
      "    cmd = 'novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f [NAME].csfasta -o SAM -F CSFASTAnQV -K [ONAME].mismatch > [ONAME].sam'\n",
      "    cmd = cmd.replace('[NAME]',csFile)\n",
      "    cmd = cmd.replace('[ONAME]',csFile.split('/')[-1].split('.')[0])\n",
      "    solidMapCmds += cmd + '\\n'\n",
      "    \n",
      "HTML(textarea('Novoalign SOLiD bash commands',solidMapCmds))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<b>Novoalign SOLiD bash commands</b>:<br>\n",
        "<textarea style=\"width: 1000px; height:200px\">\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc9_smed_gfp_w_48h_F5.csfasta -o SAM -F CSFASTAnQV -K bc9_smed_gfp_w_48h_F5.mismatch > bc9_smed_gfp_w_48h_F5.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc10_smed_gfp_7d_F3.csfasta -o SAM -F CSFASTAnQV -K bc10_smed_gfp_7d_F3.mismatch > bc10_smed_gfp_7d_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_h_72h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_h_72h_F3.mismatch > bc2_smed_reg_h_72h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_t_06h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_t_06h_F3.mismatch > bc1_smed_reg_t_06h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_h_06h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_h_06h_F3.mismatch > bc2_smed_reg_h_06h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_t_48h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_t_48h_F3.mismatch > bc2_smed_reg_t_48h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc10_smed_gfp_06h_F3.csfasta -o SAM -F CSFASTAnQV -K bc10_smed_gfp_06h_F3.mismatch > bc10_smed_gfp_06h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_hom_w_00h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_hom_w_00h_F3.mismatch > bc1_smed_hom_w_00h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_hom_w_00h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_hom_w_00h_F3.mismatch > bc2_smed_hom_w_00h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc9_smed_gfp_w_24h_F5.csfasta -o SAM -F CSFASTAnQV -K bc9_smed_gfp_w_24h_F5.mismatch > bc9_smed_gfp_w_24h_F5.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_t_36h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_t_36h_F3.mismatch > bc2_smed_reg_t_36h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc8_smed_gfp_w_5d_F3.csfasta -o SAM -F CSFASTAnQV -K bc8_smed_gfp_w_5d_F3.mismatch > bc8_smed_gfp_w_5d_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_t_24h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_t_24h_F3.mismatch > bc2_smed_reg_t_24h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_t_12h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_t_12h_F3.mismatch > bc1_smed_reg_t_12h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_h_12h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_h_12h_F3.mismatch > bc1_smed_reg_h_12h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc9_smed_hom_w_00h_F5.csfasta -o SAM -F CSFASTAnQV -K bc9_smed_hom_w_00h_F5.mismatch > bc9_smed_hom_w_00h_F5.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_h_12h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_h_12h_F3.mismatch > bc2_smed_reg_h_12h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc9_smed_gfp_w_12h_F3.csfasta -o SAM -F CSFASTAnQV -K bc9_smed_gfp_w_12h_F3.mismatch > bc9_smed_gfp_w_12h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_t_06h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_t_06h_F3.mismatch > bc2_smed_reg_t_06h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc7_smed_reg_t_48h_F3.csfasta -o SAM -F CSFASTAnQV -K bc7_smed_reg_t_48h_F3.mismatch > bc7_smed_reg_t_48h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_t_72h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_t_72h_F3.mismatch > bc1_smed_reg_t_72h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_h_48h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_h_48h_F3.mismatch > bc2_smed_reg_h_48h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_t_72h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_t_72h_F3.mismatch > bc2_smed_reg_t_72h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_h_24h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_h_24h_F3.mismatch > bc1_smed_reg_h_24h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_h_24h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_h_24h_F3.mismatch > bc2_smed_reg_h_24h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_h_72h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_h_72h_F3.mismatch > bc1_smed_reg_h_72h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc9_smed_gfp_w_48h_F3.csfasta -o SAM -F CSFASTAnQV -K bc9_smed_gfp_w_48h_F3.mismatch > bc9_smed_gfp_w_48h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_t_48h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_t_48h_F3.mismatch > bc1_smed_reg_t_48h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc12_smed_gfp_t_7d_A_F3.csfasta -o SAM -F CSFASTAnQV -K bc12_smed_gfp_t_7d_A_F3.mismatch > bc12_smed_gfp_t_7d_A_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc7_smed_reg_t_00h_A_F3.csfasta -o SAM -F CSFASTAnQV -K bc7_smed_reg_t_00h_A_F3.mismatch > bc7_smed_reg_t_00h_A_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc7_smed_reg_t_36h_F3.csfasta -o SAM -F CSFASTAnQV -K bc7_smed_reg_t_36h_F3.mismatch > bc7_smed_reg_t_36h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc9_smed_hom_w_00h_F3.csfasta -o SAM -F CSFASTAnQV -K bc9_smed_hom_w_00h_F3.mismatch > bc9_smed_hom_w_00h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc12_smed_gfp_t_7d_B_F3.csfasta -o SAM -F CSFASTAnQV -K bc12_smed_gfp_t_7d_B_F3.mismatch > bc12_smed_gfp_t_7d_B_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_t_12h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_t_12h_F3.mismatch > bc2_smed_reg_t_12h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_h_48h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_h_48h_F3.mismatch > bc1_smed_reg_h_48h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_h_36h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_h_36h_F3.mismatch > bc1_smed_reg_h_36h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc9_smed_gfp_w_24h_F3.csfasta -o SAM -F CSFASTAnQV -K bc9_smed_gfp_w_24h_F3.mismatch > bc9_smed_gfp_w_24h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc7_smed_reg_h_00h_B_F3.csfasta -o SAM -F CSFASTAnQV -K bc7_smed_reg_h_00h_B_F3.mismatch > bc7_smed_reg_h_00h_B_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc10_smed_gfp_18h_F3.csfasta -o SAM -F CSFASTAnQV -K bc10_smed_gfp_18h_F3.mismatch > bc10_smed_gfp_18h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc2_smed_reg_h_36h_F3.csfasta -o SAM -F CSFASTAnQV -K bc2_smed_reg_h_36h_F3.mismatch > bc2_smed_reg_h_36h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc10_smed_gfp_20d_smgControl_F3.csfasta -o SAM -F CSFASTAnQV -K bc10_smed_gfp_20d_smgControl_F3.mismatch > bc10_smed_gfp_20d_smgControl_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc7_smed_reg_h_00h_A_F3.csfasta -o SAM -F CSFASTAnQV -K bc7_smed_reg_h_00h_A_F3.mismatch > bc7_smed_reg_h_00h_A_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_h_06h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_h_06h_F3.mismatch > bc1_smed_reg_h_06h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_t_24h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_t_24h_F3.mismatch > bc1_smed_reg_t_24h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc9_smed_gfp_w_12h_F5.csfasta -o SAM -F CSFASTAnQV -K bc9_smed_gfp_w_12h_F5.mismatch > bc9_smed_gfp_w_12h_F5.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc1_smed_reg_t_36h_F3.csfasta -o SAM -F CSFASTAnQV -K bc1_smed_reg_t_36h_F3.mismatch > bc1_smed_reg_t_36h_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc10_smed_gfp_20d_F3.csfasta -o SAM -F CSFASTAnQV -K bc10_smed_gfp_20d_F3.mismatch > bc10_smed_gfp_20d_F3.sam\n",
        "novoalignCS -d /home/share/genome/smed/novocs/smed.novocs -f /home/share/ngsData/smed/raw/bc7_smed_reg_t_00h_B_F3.csfasta -o SAM -F CSFASTAnQV -K bc7_smed_reg_t_00h_B_F3.mismatch > bc7_smed_reg_t_00h_B_F3.sam\n",
        "\n",
        "</textarea>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 9,
       "text": [
        "<IPython.core.display.HTML at 0x1b62c50>"
       ]
      }
     ],
     "prompt_number": 9
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### Categorize and merge resulting .BAM files a common sample files\n",
      "\n",
      "Samtools merge is finicky to use. Instead, I am creating a separate `.sam` file for each grouping (lab.sex) with the correct @RG line and appending the corresponding `.bam` files to the correct header `.sam` file. This can all be done on bash. "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "bams = []\n",
      "for item in labMeta:\n",
      "    bams.append([item[0][0], item[0][2] + '.' + item[0][3]])\n",
      "    \n",
      "bams.sort(key = lambda x : x[1])\n",
      "\n",
      "bamFiles = [x for x in os.listdir('/home/share/projects/smed_strain/bwa_map') if x.split('.')[-1] == 'bam']\n",
      "\n",
      "from collections import defaultdict\n",
      "\n",
      "categories = defaultdict(list)\n",
      "for bam in bams:\n",
      "    if bam[1].find('None') == -1:\n",
      "        categories[bam[1]].append(bam[0])\n",
      "\n",
      "bwa_categorizeCmds = ''\n",
      "print 'Samples:'\n",
      "for k,v in categories.items():\n",
      "    print '    ' + k\n",
      "    meta = k.split('.')\n",
      "    #copy the common @SQ lines\n",
      "    cmd = 'cp /home/share/projects/smed_strain/header /home/share/projects/smed_strain/' + k + '.sam'\n",
      "    \n",
      "    for run in v:\n",
      "        #add specific @RG line for each run\n",
      "        rg = '@RG\\\\tID:' + run + '\\\\tSM:' + k\n",
      "        rgcmd = 'echo \"' + rg + '\" >> /home/share/projects/smed_strain/' + k + '.sam'\n",
      "        cmd += '\\n' + rgcmd\n",
      "    for run in v:\n",
      "        for bamFile in bamFiles:\n",
      "            if bamFile.find(run) != -1:\n",
      "                #append bam file to the lab.strain.sam file\n",
      "                appendcmd = 'samtools view -F 4 /home/share/projects/smed_strain/bwa_map/' + bamFile + ' >> /home/share/projects/smed_strain/' + k + '.sam'\n",
      "                cmd += '\\n' + appendcmd\n",
      "    bwa_categorizeCmds += cmd + '\\n'\n",
      "\n",
      "print '    Aboobaker.asexual'\n",
      "HTML(textarea('Categorize BWA mapping output bash commands',bwa_categorizeCmds))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Samples:\n",
        "    Sanchez.asexual\n",
        "    Bartscherer.asexual\n",
        "    Graveley.sexual\n",
        "    Rajewsky.asexual\n",
        "    Sanchez.sexual\n",
        "    Graveley.asexual\n",
        "    Pearson.asexual\n",
        "    Rink.asexual\n",
        "    Newmark.asexual\n",
        "    Aboobaker.asexual\n"
       ]
      },
      {
       "html": [
        "<b>Categorize BWA mapping output bash commands</b>:<br>\n",
        "<textarea style=\"width: 1000px; height:200px\">\n",
        "cp /home/share/projects/smed_strain/header /home/share/projects/smed_strain/Sanchez.asexual.sam\n",
        "echo \"@RG\\tID:SRR955099\\tSM:Sanchez.asexual\" >> /home/share/projects/smed_strain/Sanchez.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR955099.pe.sorted.bam >> /home/share/projects/smed_strain/Sanchez.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR955099.unpe1.sorted.bam >> /home/share/projects/smed_strain/Sanchez.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR955099.unpe2.sorted.bam >> /home/share/projects/smed_strain/Sanchez.asexual.sam\n",
        "cp /home/share/projects/smed_strain/header /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:ERR032066\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:ERR032067\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:ERR032068\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:ERR032069\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:ERR032070\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:ERR032071\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:ERR032072\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:SRR1012785\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:SRR1012831\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:SRR1012832\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:SRR1012833\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:SRR1012834\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:SRR1012835\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:SRR1012836\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "echo \"@RG\\tID:SRR1012838\\tSM:Bartscherer.asexual\" >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032066.unpe1.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032066.unpe2.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032066.pe.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032067.unpe1.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032067.unpe2.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032067.pe.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032068.pe.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032068.unpe2.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032068.unpe1.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032069.unpe1.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032069.unpe2.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032069.pe.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032070.pe.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032070.unpe2.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032070.unpe1.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032071.pe.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032071.unpe2.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032071.unpe1.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032072.unpe1.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032072.unpe2.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/ERR032072.pe.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR1012785.se.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR1012831.se.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR1012832.se.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR1012833.se.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR1012834.se.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR1012835.se.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR1012836.se.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR1012838.se.sorted.bam >> /home/share/projects/smed_strain/Bartscherer.asexual.sam\n",
        "cp /home/share/projects/smed_strain/header /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "echo \"@RG\\tID:SRR388706\\tSM:Graveley.sexual\" >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "echo \"@RG\\tID:SRR388708\\tSM:Graveley.sexual\" >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "echo \"@RG\\tID:SRR388710\\tSM:Graveley.sexual\" >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "echo \"@RG\\tID:SRR388711\\tSM:Graveley.sexual\" >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "echo \"@RG\\tID:SRR388712\\tSM:Graveley.sexual\" >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "echo \"@RG\\tID:SRR388713\\tSM:Graveley.sexual\" >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388706.se.sorted.bam >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388708.se.sorted.bam >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388710.unpe2.sorted.bam >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388710.pe.sorted.bam >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388710.unpe1.sorted.bam >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388711.se.sorted.bam >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388712.se.sorted.bam >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388713.se.sorted.bam >> /home/share/projects/smed_strain/Graveley.sexual.sam\n",
        "cp /home/share/projects/smed_strain/header /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "echo \"@RG\\tID:SRR125366\\tSM:Rajewsky.asexual\" >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "echo \"@RG\\tID:oenals4\\tSM:Rajewsky.asexual\" >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "echo \"@RG\\tID:oenals5\\tSM:Rajewsky.asexual\" >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "echo \"@RG\\tID:oenalx1\\tSM:Rajewsky.asexual\" >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "echo \"@RG\\tID:oenalx2\\tSM:Rajewsky.asexual\" >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "echo \"@RG\\tID:oenalxins\\tSM:Rajewsky.asexual\" >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR125366.se.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenals4.unpe2.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenals4.unpe1.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenals4.pe.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenals5.unpe2.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenals5.unpe1.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenals5.pe.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenalx1.pe.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenalx1.unpe1.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenalx1.unpe2.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenalx2.unpe1.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenalx2.unpe2.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenalx2.pe.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenalxins.unpe2.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenalxins.unpe1.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/oenalxins.pe.sorted.bam >> /home/share/projects/smed_strain/Rajewsky.asexual.sam\n",
        "cp /home/share/projects/smed_strain/header /home/share/projects/smed_strain/Sanchez.sexual.sam\n",
        "echo \"@RG\\tID:SRR955511\\tSM:Sanchez.sexual\" >> /home/share/projects/smed_strain/Sanchez.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR955511.unpe2.sorted.bam >> /home/share/projects/smed_strain/Sanchez.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR955511.pe.sorted.bam >> /home/share/projects/smed_strain/Sanchez.sexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR955511.unpe1.sorted.bam >> /home/share/projects/smed_strain/Sanchez.sexual.sam\n",
        "cp /home/share/projects/smed_strain/header /home/share/projects/smed_strain/Graveley.asexual.sam\n",
        "echo \"@RG\\tID:SRR388701\\tSM:Graveley.asexual\" >> /home/share/projects/smed_strain/Graveley.asexual.sam\n",
        "echo \"@RG\\tID:SRR388703\\tSM:Graveley.asexual\" >> /home/share/projects/smed_strain/Graveley.asexual.sam\n",
        "echo \"@RG\\tID:SRR388705\\tSM:Graveley.asexual\" >> /home/share/projects/smed_strain/Graveley.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388701.se.sorted.bam >> /home/share/projects/smed_strain/Graveley.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388703.se.sorted.bam >> /home/share/projects/smed_strain/Graveley.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388705.unpe2.sorted.bam >> /home/share/projects/smed_strain/Graveley.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388705.pe.sorted.bam >> /home/share/projects/smed_strain/Graveley.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR388705.unpe1.sorted.bam >> /home/share/projects/smed_strain/Graveley.asexual.sam\n",
        "cp /home/share/projects/smed_strain/header /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR496276\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR496278\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR496280\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR867380\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR867381\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR867382\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR867383\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR867384\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR867385\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR867386\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR867387\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "echo \"@RG\\tID:SRR867388\\tSM:Pearson.asexual\" >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR496276.unpe1.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR496276.unpe2.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR496276.pe.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR496278.unpe2.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR496278.pe.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR496278.unpe1.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR496280.pe.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR496280.unpe2.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR496280.unpe1.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR867380.se.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR867381.se.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR867382.se.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR867383.se.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR867384.se.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR867385.se.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR867386.se.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR867387.se.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR867388.se.sorted.bam >> /home/share/projects/smed_strain/Pearson.asexual.sam\n",
        "cp /home/share/projects/smed_strain/header /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "echo \"@RG\\tID:SRR900846\\tSM:Rink.asexual\" >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "echo \"@RG\\tID:SRR900847\\tSM:Rink.asexual\" >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "echo \"@RG\\tID:SRR900848\\tSM:Rink.asexual\" >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "echo \"@RG\\tID:SRR900849\\tSM:Rink.asexual\" >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "echo \"@RG\\tID:SRR900850\\tSM:Rink.asexual\" >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "echo \"@RG\\tID:SRR900851\\tSM:Rink.asexual\" >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "echo \"@RG\\tID:SRR900852\\tSM:Rink.asexual\" >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "echo \"@RG\\tID:SRR900853\\tSM:Rink.asexual\" >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR900846.se.sorted.bam >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR900847.se.sorted.bam >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR900848.se.sorted.bam >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR900849.se.sorted.bam >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR900850.se.sorted.bam >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR900851.se.sorted.bam >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR900852.se.sorted.bam >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR900853.se.sorted.bam >> /home/share/projects/smed_strain/Rink.asexual.sam\n",
        "cp /home/share/projects/smed_strain/header /home/share/projects/smed_strain/Newmark.asexual.sam\n",
        "echo \"@RG\\tID:SRR390350\\tSM:Newmark.asexual\" >> /home/share/projects/smed_strain/Newmark.asexual.sam\n",
        "echo \"@RG\\tID:SRR390351\\tSM:Newmark.asexual\" >> /home/share/projects/smed_strain/Newmark.asexual.sam\n",
        "echo \"@RG\\tID:SRR390352\\tSM:Newmark.asexual\" >> /home/share/projects/smed_strain/Newmark.asexual.sam\n",
        "echo \"@RG\\tID:SRR390353\\tSM:Newmark.asexual\" >> /home/share/projects/smed_strain/Newmark.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR390350.se.sorted.bam >> /home/share/projects/smed_strain/Newmark.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR390351.se.sorted.bam >> /home/share/projects/smed_strain/Newmark.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR390352.se.sorted.bam >> /home/share/projects/smed_strain/Newmark.asexual.sam\n",
        "samtools view -F 4 /home/share/projects/smed_strain/bwa_map/SRR390353.se.sorted.bam >> /home/share/projects/smed_strain/Newmark.asexual.sam\n",
        "\n",
        "</textarea>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 10,
       "text": [
        "<IPython.core.display.HTML at 0x1b92250>"
       ]
      }
     ],
     "prompt_number": 10
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#For ABI SOLiD mapping files\n",
      "#RG:Z:ERR032072\n",
      "solidFs = [x for x in os.listdir('/home/share/projects/smed_strain/novo_map/done')]\n",
      "solidFs = [x for x in solidFs if x.split('.')[-1] == 'sam']\n",
      "\n",
      "novoalign_categorizeCmds = 'samtools view -SH /home/share/projects/smed_strain/novo_map/done/' + solidFs[0] + ' > /home/share/projects/smed_strain/Aboobaker.asexual.sam'\n",
      "for f in solidFs:\n",
      "    rg = '@RG\\\\tID:' + f.split('.')[0] + '\\\\tSM:Aboobaker.asexual'\n",
      "    rgcmd = 'echo \"' + rg + '\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam'\n",
      "    novoalign_categorizeCmds += rgcmd + '\\n'\n",
      "\n",
      "for f in solidFs:\n",
      "    novoalign_categorizeCmds += 'samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/' + f + \" | sed -e 's/$/\\\\tRG:Z:\" + f.split('.')[0] + \"/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\\n\"\n",
      "    \n",
      "HTML(textarea('Categorize novoalign output bash commands',novoalign_categorizeCmds))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<b>Categorize novoalign output bash commands</b>:<br>\n",
        "<textarea style=\"width: 1000px; height:200px\">\n",
        "samtools view -SH /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_t_06h_F3.sam > /home/share/projects/smed_strain/Aboobaker.asexual.samecho \"@RG\\tID:bc2_smed_reg_t_06h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_reg_h_06h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc7_smed_reg_t_48h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_reg_t_48h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc1_smed_reg_h_36h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc1_smed_reg_t_12h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_reg_t_24h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc9_smed_gfp_w_24h_F5\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc10_smed_gfp_7d_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc9_smed_gfp_w_48h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc9_smed_hom_w_00h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc9_smed_gfp_w_24h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc1_smed_reg_t_72h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc1_smed_reg_h_48h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_reg_t_72h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc12_smed_gfp_t_7d_B_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc7_smed_reg_t_00h_A_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_reg_h_24h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc10_smed_gfp_06h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc8_smed_gfp_w_5d_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc7_smed_reg_t_36h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc1_smed_reg_h_72h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_reg_h_72h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc12_smed_gfp_t_7d_A_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc1_smed_reg_h_24h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc9_smed_hom_w_00h_F5\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_reg_t_36h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc9_smed_gfp_w_48h_F5\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_reg_h_48h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc1_smed_reg_h_12h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc1_smed_reg_t_48h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_hom_w_00h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc1_smed_hom_w_00h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_reg_t_12h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc2_smed_reg_h_12h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc1_smed_reg_t_06h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "echo \"@RG\\tID:bc9_smed_gfp_w_12h_F3\\tSM:Aboobaker.asexual\" >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_t_06h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_t_06h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_h_06h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_h_06h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc7_smed_reg_t_48h_F3.sam | sed -e 's/$/\\tRG:Z:bc7_smed_reg_t_48h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_t_48h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_t_48h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc1_smed_reg_h_36h_F3.sam | sed -e 's/$/\\tRG:Z:bc1_smed_reg_h_36h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc1_smed_reg_t_12h_F3.sam | sed -e 's/$/\\tRG:Z:bc1_smed_reg_t_12h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_t_24h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_t_24h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc9_smed_gfp_w_24h_F5.sam | sed -e 's/$/\\tRG:Z:bc9_smed_gfp_w_24h_F5/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc10_smed_gfp_7d_F3.sam | sed -e 's/$/\\tRG:Z:bc10_smed_gfp_7d_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc9_smed_gfp_w_48h_F3.sam | sed -e 's/$/\\tRG:Z:bc9_smed_gfp_w_48h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc9_smed_hom_w_00h_F3.sam | sed -e 's/$/\\tRG:Z:bc9_smed_hom_w_00h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc9_smed_gfp_w_24h_F3.sam | sed -e 's/$/\\tRG:Z:bc9_smed_gfp_w_24h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc1_smed_reg_t_72h_F3.sam | sed -e 's/$/\\tRG:Z:bc1_smed_reg_t_72h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc1_smed_reg_h_48h_F3.sam | sed -e 's/$/\\tRG:Z:bc1_smed_reg_h_48h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_t_72h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_t_72h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc12_smed_gfp_t_7d_B_F3.sam | sed -e 's/$/\\tRG:Z:bc12_smed_gfp_t_7d_B_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc7_smed_reg_t_00h_A_F3.sam | sed -e 's/$/\\tRG:Z:bc7_smed_reg_t_00h_A_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_h_24h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_h_24h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc10_smed_gfp_06h_F3.sam | sed -e 's/$/\\tRG:Z:bc10_smed_gfp_06h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc8_smed_gfp_w_5d_F3.sam | sed -e 's/$/\\tRG:Z:bc8_smed_gfp_w_5d_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc7_smed_reg_t_36h_F3.sam | sed -e 's/$/\\tRG:Z:bc7_smed_reg_t_36h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc1_smed_reg_h_72h_F3.sam | sed -e 's/$/\\tRG:Z:bc1_smed_reg_h_72h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_h_72h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_h_72h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc12_smed_gfp_t_7d_A_F3.sam | sed -e 's/$/\\tRG:Z:bc12_smed_gfp_t_7d_A_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc1_smed_reg_h_24h_F3.sam | sed -e 's/$/\\tRG:Z:bc1_smed_reg_h_24h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc9_smed_hom_w_00h_F5.sam | sed -e 's/$/\\tRG:Z:bc9_smed_hom_w_00h_F5/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_t_36h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_t_36h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc9_smed_gfp_w_48h_F5.sam | sed -e 's/$/\\tRG:Z:bc9_smed_gfp_w_48h_F5/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_h_48h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_h_48h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc1_smed_reg_h_12h_F3.sam | sed -e 's/$/\\tRG:Z:bc1_smed_reg_h_12h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc1_smed_reg_t_48h_F3.sam | sed -e 's/$/\\tRG:Z:bc1_smed_reg_t_48h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_hom_w_00h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_hom_w_00h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc1_smed_hom_w_00h_F3.sam | sed -e 's/$/\\tRG:Z:bc1_smed_hom_w_00h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_t_12h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_t_12h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc2_smed_reg_h_12h_F3.sam | sed -e 's/$/\\tRG:Z:bc2_smed_reg_h_12h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc1_smed_reg_t_06h_F3.sam | sed -e 's/$/\\tRG:Z:bc1_smed_reg_t_06h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "samtools view -S -F 4 /home/share/projects/smed_strain/novo_map/done/bc9_smed_gfp_w_12h_F3.sam | sed -e 's/$/\\tRG:Z:bc9_smed_gfp_w_12h_F3/' >> /home/share/projects/smed_strain/Aboobaker.asexual.sam\n",
        "\n",
        "</textarea>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 11,
       "text": [
        "<IPython.core.display.HTML at 0x1b7ff90>"
       ]
      }
     ],
     "prompt_number": 11
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### Post processing of mapped reads\n",
      "\n",
      "Convert and sort each sample to .bam. Use picard tools to remove duplicates. "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "consolidated = [x for x in os.listdir('/home/share/projects/smed_strain/consolidated_map/raw') if x.split('.')[-1] == 'sam']\n",
      "\n",
      "sortCmds = ''\n",
      "for conFile in consolidated:\n",
      "    name = '.'.join(conFile.split('.')[:2])\n",
      "    sortCmds += 'samtools view -bSh /home/share/projects/smed_strain/consolidated_map/raw' + conFile + ' | samtools sort - ' + name + '.sorted\\n'\n",
      "\n",
      "HTML(textarea('Sort consolidated .sam files bash commands',sortCmds))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<b>Sort consolidated .sam files bash commands</b>:<br>\n",
        "<textarea style=\"width: 1000px; height:200px\">\n",
        "samtools view -bSh /home/share/projects/smed_strain/consolidated_map/rawPearson.asexual.sam | samtools sort - Pearson.asexual.sorted\n",
        "samtools view -bSh /home/share/projects/smed_strain/consolidated_map/rawGraveley.sexual.sam | samtools sort - Graveley.sexual.sorted\n",
        "samtools view -bSh /home/share/projects/smed_strain/consolidated_map/rawAboobaker.asexual.sam | samtools sort - Aboobaker.asexual.sorted\n",
        "samtools view -bSh /home/share/projects/smed_strain/consolidated_map/rawSanchez.asexual.sam | samtools sort - Sanchez.asexual.sorted\n",
        "samtools view -bSh /home/share/projects/smed_strain/consolidated_map/rawRajewsky.asexual.sam | samtools sort - Rajewsky.asexual.sorted\n",
        "samtools view -bSh /home/share/projects/smed_strain/consolidated_map/rawSanchez.sexual.sam | samtools sort - Sanchez.sexual.sorted\n",
        "samtools view -bSh /home/share/projects/smed_strain/consolidated_map/rawNewmark.asexual.sam | samtools sort - Newmark.asexual.sorted\n",
        "samtools view -bSh /home/share/projects/smed_strain/consolidated_map/rawBartscherer.asexual.sam | samtools sort - Bartscherer.asexual.sorted\n",
        "samtools view -bSh /home/share/projects/smed_strain/consolidated_map/rawGraveley.asexual.sam | samtools sort - Graveley.asexual.sorted\n",
        "samtools view -bSh /home/share/projects/smed_strain/consolidated_map/rawRink.asexual.sam | samtools sort - Rink.asexual.sorted\n",
        "\n",
        "</textarea>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 12,
       "text": [
        "<IPython.core.display.HTML at 0x1b7f910>"
       ]
      }
     ],
     "prompt_number": 12
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "dedupCmds = ''\n",
      "\n",
      "for conFile in consolidated:\n",
      "    name = '.'.join(conFile.split('.')[:2])\n",
      "    dedupCmds += '''java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=''' + name + '''.sorted.bam OUTPUT=''' + name + '''.sorted.dedup.bam METRICS_FILE=''' + name + '''.metrics.txt\\n'''\n",
      "    \n",
      "HTML(textarea('Piccard-tools de-duplicate bash commands',dedupCmds))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<b>Piccard-tools de-duplicate bash commands</b>:<br>\n",
        "<textarea style=\"width: 1000px; height:200px\">\n",
        "java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=Pearson.asexual.sorted.bam OUTPUT=Pearson.asexual.sorted.dedup.bam METRICS_FILE=Pearson.asexual.metrics.txt\n",
        "java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=Graveley.sexual.sorted.bam OUTPUT=Graveley.sexual.sorted.dedup.bam METRICS_FILE=Graveley.sexual.metrics.txt\n",
        "java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=Aboobaker.asexual.sorted.bam OUTPUT=Aboobaker.asexual.sorted.dedup.bam METRICS_FILE=Aboobaker.asexual.metrics.txt\n",
        "java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=Sanchez.asexual.sorted.bam OUTPUT=Sanchez.asexual.sorted.dedup.bam METRICS_FILE=Sanchez.asexual.metrics.txt\n",
        "java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=Rajewsky.asexual.sorted.bam OUTPUT=Rajewsky.asexual.sorted.dedup.bam METRICS_FILE=Rajewsky.asexual.metrics.txt\n",
        "java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=Sanchez.sexual.sorted.bam OUTPUT=Sanchez.sexual.sorted.dedup.bam METRICS_FILE=Sanchez.sexual.metrics.txt\n",
        "java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=Newmark.asexual.sorted.bam OUTPUT=Newmark.asexual.sorted.dedup.bam METRICS_FILE=Newmark.asexual.metrics.txt\n",
        "java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=Bartscherer.asexual.sorted.bam OUTPUT=Bartscherer.asexual.sorted.dedup.bam METRICS_FILE=Bartscherer.asexual.metrics.txt\n",
        "java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=Graveley.asexual.sorted.bam OUTPUT=Graveley.asexual.sorted.dedup.bam METRICS_FILE=Graveley.asexual.metrics.txt\n",
        "java -jar /home/share/software/picard-tools-1.115/MarkDuplicates.jar MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=60 INPUT=Rink.asexual.sorted.bam OUTPUT=Rink.asexual.sorted.dedup.bam METRICS_FILE=Rink.asexual.metrics.txt\n",
        "\n",
        "</textarea>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 13,
       "text": [
        "<IPython.core.display.HTML at 0x1b62910>"
       ]
      }
     ],
     "prompt_number": 13
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "indexCmds = ''\n",
      "\n",
      "for conFile in consolidated:\n",
      "    name = '.'.join(conFile.split('.')[:2])\n",
      "    indexCmds += '''java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=''' + name + '''.sorted.dedup.bam\\n'''\n",
      "    \n",
      "HTML(textarea('Index sorted, deduped bash commands',indexCmds))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<b>Index sorted, deduped bash commands</b>:<br>\n",
        "<textarea style=\"width: 1000px; height:200px\">\n",
        "java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=Pearson.asexual.sorted.dedup.bam\n",
        "java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=Graveley.sexual.sorted.dedup.bam\n",
        "java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=Aboobaker.asexual.sorted.dedup.bam\n",
        "java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=Sanchez.asexual.sorted.dedup.bam\n",
        "java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=Rajewsky.asexual.sorted.dedup.bam\n",
        "java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=Sanchez.sexual.sorted.dedup.bam\n",
        "java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=Newmark.asexual.sorted.dedup.bam\n",
        "java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=Bartscherer.asexual.sorted.dedup.bam\n",
        "java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=Graveley.asexual.sorted.dedup.bam\n",
        "java -jar /home/share/software/picard-tools-1.115/BuildBamIndex.jar INPUT=Rink.asexual.sorted.dedup.bam\n",
        "\n",
        "</textarea>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 14,
       "text": [
        "<IPython.core.display.HTML at 0x1b62e10>"
       ]
      }
     ],
     "prompt_number": 14
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### Variant calling with samtools\n",
      "\n",
      "    samtools mpileup -d 10000 -uDSf /home/share/genome/smed/smed.genome.3.1.fa *.sorted.dedup.bam | bcftools view -vcg - > labs.raw.new.bcf\n",
      "    bcftools view labs.raw.bcf | vcfutils.pl varFilter -D1000 > labs.flt.vcf\n",
      "    grep -v '##' labs.flt.vcf > labs.flt.noHeader.vcf\n",
      "    "
     ]
    }
   ],
   "metadata": {}
  }
 ]
}